hack/verify_nvidia_nim_api.go

```diff
@@ -0,0 +1,95 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"github.com/opendatahub-io/odh-model-controller/controllers/utils"
+	"os"
+	"strings"
+)
+
+// Use this script for validating NVIDIA API access used by the NIM Account Controller.
+// Controller code is in: controllers/nim_account_controller.go
+// NVIDIA NIM access is encapsulated in: controllers/utils/nim.go
+//
+// This tool takes NVIDIA NGC API Key as the first positional argument.
+// It mimics the controller work by performing the following:
+//
+//  1. Fetch a list of the available runtime images from NVIDIA image registry.
+//  2. Validate the API Key by requesting an NVIDIA image registry token and using it to pull the manifest for
+//     the first item in the list of runtime images.
+//  3. Fetch model info for all the available runtimes using the API Key to request the required NGC model
+//     registry token.
+//
+// ** add -verbose to the script name to print the runtime images and models (it's a lot of data).
+func main() {
+	verbose := flag.Bool("verbose", false, "verbose")
+
+	flag.Parse()
+	if len(flag.Args()) < 1 {
+		panic("Please provide the API Key as the first positional argument")
+	}
+	apiKey := flag.Arg(0)
+
+	runtimes, rErr := utils.GetAvailableNimRuntimes()
+	if rErr != nil {
+		panic(rErr)
+	}
+	fmt.Printf("Got %d available runtimes successfully", len(runtimes))
+	fmt.Println()
+
+	if *verbose {
+		for _, runtime := range runtimes {
+			j, jErr := json.MarshalIndent(runtime, "", "  ")
+			if jErr != nil {
+				panic(jErr)
+			}
+			fmt.Println(string(j))
+		}
+	}
+
+	if vErr := utils.ValidateApiKey(apiKey, runtimes[0]); vErr != nil {
+		panic(vErr)
+	}
+	fmt.Println("API Key validated successfully")
+
+	models, dErr := utils.GetNimModelData(apiKey, runtimes)
+	if dErr != nil {
+		panic(dErr)
+	}
+	fmt.Printf("Got %d models info successfully", len(models))
+	fmt.Println()
```

Comment:
These can be combined to one print?

Reply:
Do you mean `fmt.Println(fmt.Sprintf...`?
I can try, but I think this will make _golint-ci_ mad 😁 .

Reply:
As expected, `fmt.Println(fmt.Sprintf("Got %d models info successfully", len(models)))`, breaks _golangci-lint_:

![image](https://github.com/user-attachments/assets/f6267af6-ce43-435b-ad95-dac888c304a5)

I can do: `fmt.Printf("Got %d models info successfully\n", len(models))`, but I don't usually prefer to include line breaks in string literals.

Your call.

Reply:
Resolved in https://github.com/opendatahub-io/odh-model-controller/pull/289/commits/85231d0a2756775c6a6f989c4e992f063188da4e.